dev/planning-example.py

```diff
@@ -0,0 +1,292 @@
+If no parameter is given, the current date/time is assumed.
+"""
+
+# sunset
+obs.next_sunset(date=day)
+obs.previous_sunset(date=day)
+
+# sunrise
+obs.next_sunrise(date=day)
+obs.previous_sunrise(date=day)
+
+# moon rise
+obs.next_moon_rise(date=day)
+obs.previous_moon_rise(date=day)
+
+# moon set
+obs.next_moon_set(date=day)
+obs.previous_moon_set(date=day)
+
+# The above functions can be called with an `angle` keyword argument to specify
+# a particular horizon angle for rising or setting, or can be called with
+# convenience functions for particular morning/evening twilight:
+
+obs.next_sunset(date=day, angle=18*u.degree) # (i.e., astronomical twilight)
+
+# evening (astronomical) twilight
+obs.evening_astronomical(date=day)
+
+# evening (nautical) twilight
+obs.evening_nautical(date=day)
+
+# evening (civil) twilight
+obs.evening_civil(date=day)
+
+# morning (nautical) twilight
+obs.morning_nautical(date=day)
+
+# morning (civil) twilight
+obs.morning_civil(date=day)
+
+# morning (astronomical) twilight
+obs.morning_astronomical(date=day)
+
+# what is the moon illumination?
+# returns a float, which is percentage of the moon illuminated
+obs.moon_illumination(date=t1)
+
+# what is the moon altitude and azimuth?
+# returns an altaz coordinate
+obs.moon_position(date=t1)
+
+# Other sun-related convenience functions:
+obs.noon(date=day)
+obs.midnight(date=day)
+
+# ==============
+# Target objects
+# ==============
+'''
+A target object defines a single observation target, with coordinates
+and an optional name.
+'''
+from astroplan import FixedTarget, airmass_plot
+
+# Define a target.
+t1 = FixedTarget(name='Polaris', ra='02:31:49.09', dec='+89:15:50.8')
+
+# initializing from astropy entities directly
+from astropy.coordinates import SkyCoord
+
+t1 = FixedTarget(name='Polaris',
+               coord=SkyCoord('02h31m49.09s', '+89d15m50.8s', frame='icrs'))
+
+# Leaves scope for NonSiderealTarget, etc.
+
+# Convenience methods for looking up/constructing targets by name via
+# astroquery?
+
+# Plot the airmass for the specified target:
+airmass_plot(t1, observer=obs, datetime=day)
+
+# ================================
+# Condition objects, observability
+# ================================
+
+"""
+Q: Can I observe this target on the night of May 1, 2015 between 18:30
+and 05:30 HST for 30 minutes somewhere between 15 degrees and 89 degrees
+altitude?
+"""
+
+# first we define the conditions
+from astroplan import Constraints, TimeWindow, AltitudeRange, AboveAirmass
+
+# times can be passed in as strings (interpreted as for get_date()) or
+# as astropy Time or datetime objects
+cons = [TimeWindow("2015-05-01 18:30", "2015-05-02 05:30"),
+        AltitudeRange(15.0*u.deg, 89.0*u.deg), AboveAirmass(1.2)]
+# define a target
+tgt = FixedTarget(name='S5', ra='14:20:00.00', dec='48:00:00.00')
+
+# returns an object with information about the observability with these
+# constraints
+info = obs.Constraints(tgt, cons)
```

Comment:
Any opinions on changing the name of the `Contraints` class to `CombineConstraints`? I think it will read more naturally to see: 
```python
from astroplan import CombineConstraints, TimeWindow, AltitudeWindow, AboveAirmass
constraint_list = [TimeWindow("2015-05-01 18:30", "2015-05-02 05:30"),
                   AltitudeWindow(15.0*u.deg, 89.0*u.deg), AboveAirmass(1.2)]
constraints = CombineConstraints(constraint_list)
info = constraints(obs, tgt)
```
Otherwise the word constraint will get used a billion times in singular and plural forms without any clarity about which "constraint" thing does what. @eteq @ejeschke